icd/api/vk_descriptor_set.cpp

```diff
@@ -87,6 +88,18 @@ void DescriptorSet<numPalDevices>::Reassign(
         }
     }
 
+    for (uint32_t deviceIdx = 0; deviceIdx < numPalDevices; deviceIdx++)
+    {
+        for (uint32_t bindingIndex = 0; bindingIndex < pLayout->Info().count; ++bindingIndex) {
+          const DescriptorSetLayout::BindingInfo& bindingInfo = pLayout->Binding(bindingIndex);
+
+          if (bindingInfo.imm.dwSize == 0) continue;
+
+          uint32_t * pSamplerDesc =(pLayout->Info().imm.pImmutableSamplerData + bindingInfo.imm.dwOffset);
+          uint32_t* pDestAddr = StaticCpuAddress(deviceIdx) + Layout()->GetDstStaOffset(bindingInfo, 0);
+          memcpy(pDestAddr, pSamplerDesc, 4*bindingInfo.imm.dwSize);
```

Comment:
Writing the sampler in when assigning the layout to the descriptor set makes the most sense to me.  However, it does require the device to be passed in.

Reply:
I don't see where you're using device in here.

Reply:
Use sizeof(uint) instead of 4.

Reply:
@s-perron please change 4 to sizeof(uint) ( or maybe sizeof(uint32_t)? )